templates/install_puppet.bat.erb

```diff
@@ -15,7 +15,15 @@ FOR /F "tokens=*" %%A IN ('tasklist /FI "PID eq %AGENT_PID%" /NH') DO set _task=
 echo %_task% | findstr "No tasks are running" >nul
 IF NOT %errorlevel% == 0 ( GOTO wait_for_pid )
 
-start /wait msiexec.exe /qn /norestart /i "<%= @_msi_location %>" /l*v "<%= @_logfile %>" PUPPET_MASTER_SERVER="<%= @_puppet_master %>"
+REM This *must* occur after Puppet Agent has finished applying its
+REM prior catalog which manages the pxp-agent service state. If not,
+REM the catalog includes the PE module which starts the service and
+REM sets its startup type, which prevents installs from proceeding.
+REM This may fail on agents without pxp-agent, but since this is not
+REM run interactively and the next command sets ERRORLEVEL, it's OK.
+net stop pxp-agent
```

Comment:
Did you determine the issue? 

Also keep in mind that 3.x agents upgrading won't have this - so if it fails that should be ignored.